xarray/conventions.py

```diff
@@ -165,11 +165,18 @@ def decode_cf_timedelta(num_timedeltas, units):
     """
     num_timedeltas = _asarray_or_scalar(num_timedeltas)
     units = _netcdf_to_numpy_timeunit(units)
+
+    shape = None
+    if isinstance(num_timedeltas, np.ndarray):
```

Comment:
I would unilaterally convert to a numpy array: `num_timedeltas = np.asarray(num_timedeltas)`. Then save the shape and ravel. Note that `.ravel()` is basically free for 1D data since it only makes a copy if necessary.